Main.hs

```diff
@@ -4,23 +4,40 @@ import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+short (a : b : c : xs) = False
+short _ = True
 
 lovely :: [Int] -> Bool
-lovely = undefined 
+lovely a = short a || a!!2 == 14
 
 rightTriangles :: [(Int, Int, Int)]
-rightTriangles = undefined 
+rightTriangles = [ (a, b, c) | c <- [1..], b <- [1..c], a <-[1..b], a^2 + b^2 == c^2]
 
 fizzBuzz :: [String]
-fizzBuzz = undefined 
+fizzBuzz = [s | i <- [1 .. ], let s | i `mod` 3 == 0 && i `mod` 5 == 0 = "FizzBuzz"
+                                    | i `mod` 3 == 0 = "Fizz"
+                                    | i `mod` 5 == 0 = "Buzz"
```

Comment:
Avoid code duplication. Either check for divisibility by 15, or introduce functions for divisibility by 3 and 5 respectively.